app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,39 @@
+class SessionsController < ApplicationController
+
+  def new
+  end
+
+  def create
+    user = User.find_by(username: session_params[:username])
+    # If the user exists AND the password entered is correct.
+    if user && user.authenticate(session_params[:password])
+      # Save the user id inside the browser cookie. This is how we keep the user
+      # logged in when they navigate around our website.
+      session[:user_id] = user.id
```

Comment:
This is fine for now. As we get more features built, how are we planning to handle session data? Persist all data as cookies or perhaps persist a session kay and keep the rest of the data on the server? No need to answer - just stuff to think about.

Reply:
Will do. I do like the idea of saving a "state" object in the db, and then just saving it's key in the session - much more private, less storage. The session object can't hold too much.